src/test/java/com/thealgorithms/strings/PermuteStringTest.java

```diff
@@ -0,0 +1,36 @@
+package com.thealgorithms.strings;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class PermuteStringTest {
+
+    private Set<String> getPermutations(String str) {
+        Set<String> permutations = new HashSet<>();
+        PermuteString.generatePermutations(str, 0, str.length(), permutations);
+        return permutations;
+    }
```

Comment:
Could you move this method to `PermuteString` and make the existing `generatePermutations` private? To avoid mutating function arguments and provide a cleaner interface

Reply:
Thanks, this improve class design. Fixed.